src/iterator.cr

```diff
@@ -293,34 +293,30 @@ module Iterator(T)
   # iter.next # => Iterator::Stop::INSTANCE
   # ```
   #
-  # By default, a new array is created and yielded for each consecutive when invoking `next`.
+  # By default, a new array is created and returned for each consecutive call of `next`.
   # * If *reuse* is given, the array can be reused
-  # * If *reuse* is an `Array`, this array will be reused
-  # * If *reuse* is truthy, the method will create a new array and reuse it.
+  # * If *reuse* is `true`, the method will create a new array and reuse it.
+  # * If *reuse*  is an instance of `Array`, `Deque` or a similar collection type (implementing `#<<`, `#shift` and `#size`) it will be used.
+  # * If *reuse* is falsey, the array will not be reused.
   #
   # This can be used to prevent many memory allocations when each slice of
   # interest is to be used in a read-only fashion.
   def cons(n : Int, reuse = false)
```

Comment:
This argument should also be called `count` to line up with `Enumerable#each_cons`. But that would be a breaking change (and I don't think the other changes are breaking), so it can be left for another PR.

Reply:
Pedantic: there's a double space in the comment above.

Reply:
I noticed it, too, but didn't want to be too pedantic :P

Reply:
@Sija, that would be breaking. Let's change it in another PR.